### PR TITLE
DCAC-198: Add ch- prefixed resource to each image bucket policy statement

### PR DESCRIPTION
**Pull request by VaughanJackson123**
* Further testing today confirmed our impression that one of the current issues with the `staging` deployment of the `document-signing-api` is that whilst is has read access to the Document API(?) documents held on `s3://document-api-images-cidev`, it does not have read access to the Document API(?) documents held on `s3://ch-document-api-images-staging`.
* I speculate the appropriate read permissions may be extended to the staging image bucket as well by adding extra resources that hopefully match against the staging image bucket name with its `ch-` prefix.

### CODE REVIEW

File: terraform/groups/document-signing-api/data.tf [hunk 1]
```diff
@@ -9,7 +9,8 @@ data "aws_iam_policy_document" "image_bucket" {
     ]
 
     resources = [
-      "arn:aws:s3:::${var.image_bucket_name_prefix}-*"
+      "arn:aws:s3:::${var.image_bucket_name_prefix}-*",
+      "arn:aws:s3:::ch-${var.image_bucket_name_prefix}-*"
```
Comment (marcransome) [inline terraform/groups/document-signing-api/data.tf:13]: The changes here will result in an unnecessary resource in the policy which could be cause for confusion—`document-api-images-*` in the `staging` account won't match any resources, and similarly `ch-document-api-images-*` in the `development` account won't match any resources[^1].

Revert these changes:

```suggestion
      "arn:aws:s3:::${var.image_bucket_name_prefix}-*"
```

Then, override the `image_bucket_name_prefix` variable default value for the `staging` environment in the `groups/document-signing-api/profiles/staging-eu-west-2/vars` file:

```terraform
image_bucket_name_prefix = "ch-document-api-images"
```

[^1]: A single bucket in the `development` account _does_ include the `ch-` prefix, however this should be ignored as it was created for testing iXBRL and is outdated.

File: terraform/groups/document-signing-api/data.tf [hunk 2]
```diff
@@ -23,7 +24,8 @@ data "aws_iam_policy_document" "image_bucket" {
     ]
 
     resources = [
-      "arn:aws:s3:::${var.image_bucket_name_prefix}-*/*"
+      "arn:aws:s3:::${var.image_bucket_name_prefix}-*/*",
+      "arn:aws:s3:::ch-${var.image_bucket_name_prefix}-*/*"
```
Comment (marcransome) [inline terraform/groups/document-signing-api/data.tf:28]: As above, revert these changes. Overriding the `image_bucket_name_prefix` variable default value should be enough:

```suggestion
      "arn:aws:s3:::${var.image_bucket_name_prefix}-*/*"
```

**Review by marcransome — CHANGES_REQUESTED** [review]

File: terraform/groups/document-signing-api/data.tf (hunk 1 above)
Comment (VaughanJackson123) [reply terraform/groups/document-signing-api/data.tf]: Thank you @marcransome for explaining this. I have made the changes.

File: terraform/groups/document-signing-api/data.tf (hunk 2 above)
Comment (VaughanJackson123) [reply terraform/groups/document-signing-api/data.tf]: Made this change too.

**Review by marcransome — APPROVED** [review]
